### PR TITLE
add updated hub client

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

**Comment by danewalton** [comment]
**Note to self make sure to add files to CMake once the json is merged**

**Comment by danewalton** [comment]
[Code cov stands at 86%](https://github.com/Azure/azure-iot-middleware-freertos/pull/133/checks?check_run_id=2748725882#step:8:269)

**Review by hihigupt — APPROVED** [review]
:shipit:

**Review by jspaith — APPROVED** [review]

File: source/azure_iot_hub_client.c [hunk 1]
```diff
@@ -351,17 +353,17 @@ static uint32_t prvAzureIoTHubClientDeviceTwinProcess( AzureIoTHubClientReceiveC
...
-                xTwinResponse.ulRequestID = ulRequestID;
-
-                AZLogDebug( ( "Invoking twin callback" ) );
-                pxContext->_internal.callbacks.xTwinCallback( &xTwinResponse,
-                                                              pxContext->_internal.pvCallbackContext );
-                AZLogDebug( ( "Returning from twin callback" ) );
+                xPropertiesResponse.pvMessagePayload = xMQTTPublishInfo->pvPayload;
+                xPropertiesResponse.ulPayloadLength = ( uint32_t ) xMQTTPublishInfo->xPayloadLength;
+                xPropertiesResponse.xMessageStatus = ( AzureIoTHubMessageStatus_t ) xOutRequest.status;
+                /* xPropertiesResponse.pucVersion = az_span_ptr( xOutRequest.version ); */
```
[the remark this answers is not in the document]
Comment (danewalton) [reply source/azure_iot_hub_client.c:359]: crap good catch thanks

File: source/azure_iot_hub_client.c [hunk 2]
```diff
@@ -1181,42 +1185,42 @@ AzureIoTResult_t AzureIoTHubClient_SubscribeDeviceTwin( AzureIoTHubClient_t * px
...
+            AZLogError( ( "Properties subscribe failed: MQTT error=0x%08x", xMQTTResult ) );
             xResult = eAzureIoTErrorSubscribeFailed;
         }
         else
         {
             pxContext->_internal.usState = azureiothubTOPIC_SUBSCRIBE_STATE_SUB;
             pxContext->_internal.usMqttSubPacketID = usSubscribePacketIdentifier;
-            pxContext->_internal.pxProcessFunction = prvAzureIoTHubClientDeviceTwinProcess;
-            pxContext->_internal.callbacks.xTwinCallback = xCallback;
+            pxContext->_internal.pxProcessFunction = prvAzureIoTHubClientDevicePropertiesProcess;
```
[the remark this answers is not in the document]
Comment (danewalton) [reply source/azure_iot_hub_client.c:1216]: Changed here and everywhere. So for example:
-  `AzureIoTHubClient_SendDevicePropertiesReported` -> `AzureIoTHubClient_SendPropertiesReported`
- `AzureIoTHubClient_GetDeviceProperties` -> `AzureIoTHubClient_GetProperties`

**Pull request merged**